src/modules/navigator/navigator_main.cpp

```diff
@@ -835,12 +834,10 @@ void Navigator::geofence_breach_check(bool &have_geofence_position_data)
 
 		if (_geofence.getPredict()) {
 			fence_violation_test_point = _gf_breach_avoidance.getFenceViolationTestPoint();
-			snprintf(geofence_violation_warning, sizeof(geofence_violation_warning), "Approaching on geofence");
 
 		} else {
 			fence_violation_test_point = matrix::Vector2d(_global_pos.lat, _global_pos.lon);
 			vertical_test_point_distance = 0;
-			snprintf(geofence_violation_warning, sizeof(geofence_violation_warning), "Geofence exceeded");
```

Comment:
Here's I added a way to distinguish between the GF_PREDICT options to provide accurate feedback. Your change removes this distinction.

Reply:
I removed it as it wasn't correct. The events call requires a string literal.
I passed the reason along and report it at the common place: https://github.com/PX4/PX4-Autopilot/pull/20172/commits/209f6accf7f5892479f40c06179d9c6b33955d44